src/Connections/GenericHook.ts

```diff
@@ -107,27 +111,26 @@ export class GenericHookConnection extends BaseConnection implements IConnection
         return obj;
     }
 
-    static validateState(state: Record<string, unknown>, allowJsTransformationFunctions?: boolean): GenericHookConnectionState {
+    static validateState(state: Record<string, unknown>): Omit<GenericHookConnectionState, "hookId"> {
```

Comment:
This feels a bit weird, why not just make hookId optional in the state as it was originally?

Reply:
This was part of making `hookId` a mandatory property, lest it's not set in some state update/creation functions.  Not sure how important that is, so I can revert that if needed.

Reply:
Reverted